server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -651,7 +666,10 @@ public final class IndexSettings {
     private volatile ByteSizeValue generationThresholdSize;
     private volatile ByteSizeValue flushAfterMergeThresholdSize;
     private final MergeSchedulerConfig mergeSchedulerConfig;
-    private final MergePolicyConfig mergePolicyConfig;
+    private final TieredMergePolicyProvider tieredMergePolicyProvider;
+    private final LogByteSizeMergePolicyProvider logByteSizeMergePolicyProvider;
+    private final MergePolicyProvider defaultMergePolicyProvider;
```

Comment:
Why do you need these fields? I think the only place where you need to make a decision `getMergePolicy`:

```
public MergePolicy getMergePolicy(boolean isTimeIndex) {
  // check IndexScopedSettings
  // if `isTimeIndex` check Settings for default
  // return merge policy
}
```

Reply:
this has simplified the code a lot although we still need `tieredMergePolicyProvider` and `logByteSizeMergePolicyProvider` to register their settingsUpdateConsumer.